@LinProp/LinProp.m

```diff
@@ -50,6 +63,42 @@
     end
     methods
         function obj = LinProp(varargin)
+            % LinProp Const:
+            % a = LinProp(value)
+            %
+            % LinProp Input (RealUncNumber)
+            % a = LinProp(value, standard_unc, [idof])
+            %
+            % LinProp Input (RealUncNumber)
+            % a = LinProp(value, standard_unc, description)
+            %
+            % LinProp Input (ComplexUncNumber)
+            % a = LinProp(value, (covariance), [description])
+            %
+            % LinProp Input (RealUncArray)
+            % [a] = LinProp((value), (covariance), [description])
+            %
+            % LinProp Input (ComplexUncArray)
+            % [a] = LinProp((value), (covariance), [description])
+            %
+            % LinProp From Samples
+            % a = LinProp((samples), 'samples', [description], [probability])
+            %
+            % LinProp Xml String
+            % a = LinProp(xml_string)
+            %
+            % LinProp Xml File
+            % a = LinProp(filepath, 'xml_file')
+            %
+            % LinProp Binary File
+            % a = LinProp(filepath, 'binary_file')
+            %
+            % LinProp System (RealUncNumber)
+            % a = LinProp(value, (sys_inputs), (sys_sensitivities), 'system')
```

Comment:
This works only LinProp objects. I think we should mention that at least in the documentation.

At some point we have to think about changing the code, but then the wrappers for LinProp, DistProp and MCProp are not the same anymore. Perhaps a better idea would be that we come up with common base class and LinProp, DistProp and MCProp inherits from the common base class.

What do you think about it?

Reply:
> At some point we have to think about changing the code, but then the wrappers for LinProp, DistProp and MCProp are not the same anymore. Perhaps a better idea would be that we come up with common base class and LinProp, DistProp and MCProp inherits from the common base class.
> 
> What do you think about it?

I have also thought about that a few times. In theory, I think it would be good to have an abstract(?) base class, which provides common functionalities. There are quite a lot places where `LinProp` could be replaced with a generic term or statement. Matlab for example allows calling static methods with `object.method(...)`, which calls the static method `method` of the class of `object`. However, there are also a lot of places where I currently do not see how to prevent the parallel implementations in an efficient manner, especially all the places where static .NET methods are called and all methods that create new *Prop objects. Still, it might be worth to further investigate such a change.

For now, it might be best so simply insert a hook into the constructor and have that hook handle the 'system' call. If that hook were in a seprarte file, it could be different for the different classes. I have already modified (hacked?) the `_Copy_LinProp_to_DistProp_and_MCProp.cmd` script to not modify the string 'LinProp only', so the documentation could easily represent this special case.

Reply:
I have added a note in the documentation of the constructor, but not modified any of the code. (I think that would be best in a separte branch/pull-request.)